### PR TITLE
Use Vue-I18n for language translations

### PR DESCRIPTION
**Pull request by Muhaddil**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,9 @@
     "jszip": "^3.10.1",
     "pinia": "^2.1.6",
     "sass": "^1.64.1",
-    "vue": "^3.3.4"
+    "vue": "^3.3.4",
+    "vue-i18n": "^9.8.0",
+    "vue-router": "^4.2.5"
```
Comment (Lenni009) [inline package.json:22]: `vue-router` is no longer needed
```suggestion
```

File: .env
Comment (Lenni009) [inline .env:1]: These environment variables are all unused now -> The file can be deleted

File: src/main.js
Comment (Lenni009) [inline src/main.js:1]: There should be no harm in renaming this to `.ts` to have type safety

File: index.html [hunk 2]
```diff
@@ -8,7 +8,7 @@
     />
     <script
       type="module"
-      src="/src/main.ts"
+      src="/src/main.js"
```
Comment (Lenni009) [inline index.html:11]: Using TypeScript should be preferred for the type safety
```suggestion
      src="/src/main.ts"
```

File: tsconfig.app.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
     "target": "ESNext",
     "useDefineForClassFields": true,
     "module": "ESNext",
+    "allowJs": true,
```
Comment (Lenni009) [inline tsconfig.app.json:9]: The codebase should ideally be fully type safe, and therefore not contain any JS
```suggestion
```

File: tsconfig.json [hunk 4]
```diff
@@ -1,5 +1,9 @@
 {
   "files": [],
+  "compilerOptions": {
+    "allowJs": true,
```
Comment (Lenni009) [inline tsconfig.json:4]: Codebase should be fully type safe without JS
```suggestion
```

File: src/components/NavBarThemeSwitch.vue
Comment (Lenni009) [inline src/components/NavBarThemeSwitch.vue:1]: This component seems unused...? -> Delete if unused

**Review by Lenni009 — CHANGES_REQUESTED** [review]
Much much better than the last PR! Only some minor changes required, then it's fine to merge :D

File: src/components/NavBarThemeSwitch.vue
Comment (Muhaddil) [reply src/components/NavBarThemeSwitch.vue:1]: It's unused

File: index.html (hunk 2 above)
Comment (Muhaddil) [reply index.html]: Done

File: package.json (hunk 1 above)
Comment (Muhaddil) [reply package.json]: Done

**Comment by Lenni009** [comment]
LGTM, thanks!

**Pull request merged**